packages/typescript-estree/src/create-program/createWatchProgram.ts

```diff
@@ -270,6 +270,14 @@ function createWatchProgram(
     /*reportWatchStatus*/ () => {},
   ) as WatchCompilerHostOfConfigFile<ts.BuilderProgram>;
 
+  watchCompilerHost.realpath = (path: string): string => {
+    try {
+      return fs.realpathSync(path);
```

Comment:
I'm not sure how this would fix the issue as this is exactly the same API that calls by default.

We provide no implementation, so it uses the default "system" `realpath`:
https://github.com/microsoft/TypeScript/blob/988fa85f88d581e5012fd50a3621bff69c3edb2c/src/compiler/watch.ts#L632

Which is `fs.realPathSync`:
https://github.com/microsoft/TypeScript/blob/988fa85f88d581e5012fd50a3621bff69c3edb2c/src/compiler/sys.ts#L1286

Reply:
Yes, `tsc` uses this default implementation too. But I debugged and found that `host.realpath` is undefined when running eslint. Seems that typescript-eslint/estree doesn't "have" this default implementation. No idea what's really happening here.
<img width="781" alt="image" src="https://user-images.githubusercontent.com/1356263/174974030-559c45f5-b56e-402e-a3f2-f9193b9c5c80.png">

Reply:
I found the opposite to be true when debugging a lint run.

This needs further investigation.